chain/consensus/hierarchical/types.go

```diff
@@ -78,3 +68,38 @@ var _ abi.Keyer = SubnetKey("")
 func (id SubnetKey) Key() string {
 	return string(id)
 }
+
+func CommonParent(from, to address.SubnetID) (address.SubnetID, int) {
+	s1 := strings.Split(from.String(), "/")
+	s2 := strings.Split(to.String(), "/")
+	if len(s1) < len(s2) {
+		a := s1
```

Comment:
why not s1, s2 = s2, s1 ?

Reply:
Good catch! I forgot to change it after doing some tests. Thanks!